set.mm

```diff
@@ -67010,6 +67010,44 @@ more bytes (305 versus 282).  (Contributed by AV, 3-Feb-2021.)
     df-f1o sylibr ) ABGDHEAIFAIEFJKLZACDMZNZACDOZNZPZACDQZNUDUFUHABCDEFRSUEUGLU
     IUJUEUGTACDUBUAUC $.
 
+  $( Swapping two values in a bijection between two classes yields another
+     bijection between those classes.  (Contributed by BTernaryTau,
+     26-Aug-2024.) $)
+  f1ofvswap $p |- ( ( F : A -1-1-onto-> B /\ X e. A /\ Y e. A ) ->
+      ( ( F |` ( A \ { X , Y } ) ) u.
+      { <. X , ( F ` Y ) >. , <. Y , ( F ` X ) >. } ) : A -1-1-onto-> B ) $=
```

Comment:
A more conceptual proof is to note that your function is the composition of the permutation of X and Y in A followed by the original bijection, and as such is a bijection.

To prove that the first permutation is a bijection, prove that it is an involution, that is, it is its own inverse.

Reply:
That's a good point. I initially ruled out working with compositions since there isn't yet a nice way to prove that they exist without using ax-pow, but f1ofvswap doesn't require existence so that wouldn't be an issue.

Reply:
I used f1oprswap and f1oun instead of showing that the whole permutation is an involution, but otherwise the new proof should match your approach (while also being much shorter than what I had before).

Reply:
Yes, that's it, and it's indeed much shorter.